test-data/unit/check-varargs.test

```diff
@@ -592,3 +592,23 @@ class C:
     def foo(self) -> None: pass
 C().foo()
 C().foo(1)  # The decorator's return type says this should be okay
+
+
+[case testInvariantDictArgNote]
+from typing import Dict, Sequence
+def f(a: Dict[str, Sequence[int]]) -> None: pass
+a = {'a': [1, 2]}
+f(a) # E: Argument 1 to "f" has incompatible type Dict[str, List[int]]; expected Dict[str, Sequence[int]] \
+# N: "Dict" is invariant --- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance \
+# N: Consider using "Mapping" instead, which is covariant in the value
+[builtins fixtures/dict.pyi]
+
+
+[case testInvariantListArgNote]
+from typing import List, Union
+def f(numbers: List[Union[int, float]]) -> None: pass
+a = [1, 2]
+f(a) # E: Argument 1 to "f" has incompatible type List[int]; expected List[Union[int, float]] \
+# N: "List" is invariant --- see http://mypy.readthedocs.io/en/latest/common_issues.html#variance \
+# N: Consider using "Sequence" instead, which is covariant
+[builtins fixtures/list.pyi]
```

Comment:
Add tests:
* for tricky names (like ``DictReader``)
* for unrelated list types (like ``List[str]`` vs ``List[int]``), see my comment above
* for situations where covariance will help, and where it will not (like ``Dict[str, float]`` vs ``Dict[str, int]`` and vice-versa).

In general, it is a good idea to add tests for all situations that appeared in the discussion of a PR.